apps/emqx/test/emqx_hooks_SUITE.erl

```diff
@@ -37,7 +44,30 @@ all() -> emqx_common_test_helpers:all(?MODULE).
 
 % t_add(_) ->
 %     error('TODO').
-    
+
+t_add_hook_order(_) ->
+    ?assert(proper:quickcheck(add_hook_order_prop(), [{on_output, fun ct:print/2}])).
+
+add_hook_order_prop() ->
+    %% Note: order is inversed, since higher prio hooks run first:
+    Comparator = fun({Prio1, M1, F1}, {Prio2, M2, F2}) ->
+                         Prio1 > Prio2 orelse
+                             (Prio1 =:= Prio2 andalso {M1, F1} =< {M2, F2})
+                 end,
+    ?FORALL(
+       Hooks, list({range(-1, 5), atom(), atom()}),
+       try
+           {ok, _} = emqx_hooks:start_link(),
+           [emqx:hook(prop_hook, {M, F, []}, Prio) || {Prio, M, F} <- Hooks],
+           Callbacks = emqx_hooks:lookup(prop_hook),
+           Order = [{Prio, M, F} || {callback, {M, F, _}, _Filter, Prio} <- Callbacks],
```

Comment:
Q: in the [`add_callback`](https://github.com/emqx/emqx/pull/7086/files#diff-6845a349eb4566742e73f3a17bd8caf8507cc782d33934d2db26934a547947beR302) function, the `action` is named `MFA`, which gives the impression that the arguments are being used for the order as well.

Should they be generated in the property as well?

Reply:
I was thinking about that, but using a list generator will misguide proper into exploring different lists, instead of generating more relevant testcases with more hooks with different priorities. Since we know pretty well that Erlang list sorting works, I skipped that part for simplicity.